grype/matcher/java/matcher.go

```diff
@@ -21,5 +125,44 @@ func (m *Matcher) Type() match.MatcherType {
 }
 
 func (m *Matcher) Match(store vulnerability.Provider, d *distro.Distro, p pkg.Package) ([]match.Match, error) {
-	return search.ByCriteria(store, d, p, m.Type(), search.CommonCriteria...)
+	matches := make([]match.Match, 0)
+	if m.SearchMavenUpstream {
+		upstreamMatches, err := m.matchUpstreamMavenPackages(store, p)
+		if err != nil {
+			return nil, fmt.Errorf("failed to match upstream maven data")
```

Comment:
Following up on an earlier comment, it looks like a single SHA that isn't found will skip what could be valid matches without the maven data since it isn't executing the stuff below?